### PR TITLE
Added eslint custom rule to encourage address literals to be lowercase

### PR DESCRIPTION
**Pull request by jaimehgb**
I’ve been having issues because of the casing of addresses. Am I the only one? 😅 

When we get an `Address` anywhere in our mappings, if we call `toHexString()` it returns the address all in lowercase (not checksumed). We usually do address comparison using their string values instead of `Address.equals` or `Address.notEqual`. 

This creates scenarios where `SomeEntity.id = SOME_CONSTANT_ADDRESS`, which if we happen to compare `event.address.toHexString() == SomeEntity.id` it won’t match. This might lead to duplicate entities (one with the lowercase ID and one with the checksumed ID), events going unhandled if some entity is missing (`Market.load(event.address.toHexString()` when ID is checksumed), and so on. There are a couple ways of preventing this:
- Enforce all our hardcoded constants to always be lowercase
- Enforce all address comparisons to be done with `Address.equals` instead of `toHexString() == …`

I’ve created a simple eslint rule to check for the first option (easier to implement). Also, the second option might still lead to errors if we do something like:
```
if (event.address.equals(Address.fromString(SOME_CONSTANT)) {
  // do something
  Entity.load(SOME_CONSTANT);
}
...
// And then somewhere else
Entity.load(event.address.toHexString();
```

### Things against

Having the linter enforce addresses to be lowercase will probably cause issues in already existing subgraphs if we change a constant and graft. Because if some entity ID is checksumed and we change all the code logic to now deal with lowercase addresses there will be a mismatch there. It should be safe to enforce on new subgraphs, and subgraphs we index from scratch. It might be reasonable to only enable this rule for currently non-existing files somehow. Let me know your thoughts on this.

### CODE REVIEW

**Comment by dmelotik** [comment]
I have ran into this, I usually opt to compare `Address` types to prevent this

**Comment by tnkrxyz** [comment]
I ran into this mysel. Very helpful improvement.

In terms of potential impact to grafting of existing subgraphs, is there a way we can know beforehand whether grafting works after changing an address constant?

File: subgraphs/_eslint-rules/index.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+module.exports = {
+    rules: {
+        "no-checksum-addresses": require("./no-checksum-addresses"),
+    },
+};
```
Comment (dmelotik) [inline subgraphs/_eslint-rules/index.js:6]: add newline at EOF

File: subgraphs/_eslint-rules/no-checksum-addresses.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                        message: 'Ethereum addresses should be lowercase',
+                        fix: function (fixer) {
+                            return fixer.replaceText(node, node.raw.toLowerCase());
+                        },
+                    });
+                }
+            },
+        };
+    }
+}
```
Comment (dmelotik) [inline subgraphs/_eslint-rules/no-checksum-addresses.js:27]: new line at EOF

**Review by dmelotik — COMMENTED** [review]
> Because if some entity ID is checksumed and we change all the code logic to now deal with lowercase addresses there will be a mismatch there.

This is a good point as `ID` type id's are case sensitive. But in the case we graft and it doesn't work we just ignore the lint rule imo. There shouldn't be too many cases of this.

**Comment by dmelotik** [comment]
@jaimehgb can you do a test file to see if it works properly?

**Comment by jaimehgb** [comment]
Works :D 
Screenshot 2023-02-03 at 21 29 37

**Review by dmelotik — APPROVED** [review]
Nice!

**Pull request merged**